File: SuperzSpark/src/main/java/com/github/superzhc/spark/WordCount.java
package com.github.superzhc.spark;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function2;
import org.apache.spark.api.java.function.PairFunction;
import scala.Tuple2;

import java.util.Arrays;

/**
 * 2020年07月02日 superz add
 */
public class WordCount
{
    public static void main(String[] args) {
        SparkConf conf = new SparkConf().setAppName("demo").setMaster("local");
        JavaSparkContext sc = new JavaSparkContext(conf);

        JavaRDD<String> rdd = sc.textFile(WordCount.class.getClassLoader().getResource("data/wordcount.txt").getPath());
        JavaPairRDD<String, Integer> rdd2 = rdd.flatMap(d -> Arrays.asList(d.split(" ")).iterator())
                .mapToPair(d -> new Tuple2<>(d, 1));
//                .mapToPair(new PairFunction<String, String, Integer>()
//                {
//                    @Override
//                    public Tuple2<String, Integer> call(String s) throws Exception {
//                        return new Tuple2<>(s, 1);
//                    }
//                });

        rdd2 = rdd2.reduceByKey((x, y) -> x + y);
        rdd2.foreach(d -> System.out.printf("(%s,%d)\n", d._1, d._2));
    }
}

File: superz-hadoop/superz-hadoop-spark/src/main/scala/com/github/superzhc/hadoop/spark/scala/WordCount.scala
package com.github.superzhc.hadoop.spark.scala

import org.apache.spark.SparkConf
import org.apache.spark.sql.SparkSession

object WordCount {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setAppName("word count").setMaster("local")
    val spark = SparkSession.builder().config(conf).getOrCreate()
    val sc = spark.sparkContext
    // 设置Spark程序运行时的日志等级
    sc.setLogLevel("warn")
    val rdd = sc.textFile("D://data.txt")
    val wc = rdd.flatMap(_.split(" ")).map((_, 1)).reduceByKey(_ + _)
    wc.cache()
    val max = wc.map(_._2).max()
    val maxWorldRdd = wc.filter(_._2 == max)

    import spark.implicits._
    val df = maxWorldRdd.toDF("world", "nums")
    df.write.parquet("test.parquet")
    spark.stop()
  }
}
